File: 1742-maximum-number-of-balls-in-a-box/1742-maximum-number-of-balls-in-a-box.java
class Solution {
    public int countBalls(int lowLimit, int highLimit) {
        int[] arr=new int[46];

        for(int i=lowLimit;i<=highLimit;i++) {
            int a=i;
            int sum=0;
            while(a!=0)
            {
                sum+=a%10;
                a/=10;
            }
            arr[sum]++;
        }

        int result=0;

        for(int i=0;i<46;i++) {
            if(arr[i]>result)
                result=arr[i];
        }

        return result;
    
    }
}

File: 1742-maximum-number-of-balls-in-a-box/1742-maximum-number-of-balls-in-a-box.py
class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        ball_map = collections.defaultdict(int)
        for num in range(lowLimit, highLimit+1):
            num = sum([int(x) for x in str(num)])
            ball_map[num] += 1
        return max(ball_map.values())